var/spack/repos/builtin/packages/glib/package.py

```diff
@@ -144,8 +146,9 @@ class Glib(MesonPackage, AutotoolsPackage):
     depends_on("zlib-api")
     depends_on("gettext")
     depends_on("perl", type=("build", "run"))
-    # Uses distutils in gio/gdbus-2.0/codegen/utils.py
+    # Uses distutils (@:2.78) or packaging (@2.79:) in gio/gdbus-2.0/codegen/utils.py
     depends_on("python@:3.11", type=("build", "run"), when="@2.53.4:")
```

Comment:
Would `glib@2.79:` still depends on `python@:3.11`? It should be able to be built with a higher-versioned python. Using 
```python
    depends_on("python@:3.11", type=("build", "run"), when="@2.53.4:2.78")
```
may be better.

Reply:
Good point. I'll evaluate when I have a chance to get back to this PR.

Reply:
See #47766 for the correct Python version constraints